src/mem/epoch/markable_atomic.rs

```diff
@@ -0,0 +1,220 @@
+use std::marker::PhantomData;
+use std::mem;
+use std::ptr;
+use std::sync::atomic::{self, Ordering};
+
+use super::{Owned, Shared, Guard};
+
+/// Like a `(std::sync::atomic::AtomicPtr, bool)` pair.
```

Comment:
This description is a bit inconsistent: the pointer is atomic and mutable, while the boolean in not. I think this line should either be cut (leaving the more precise description below) or be consistent, saying either `(AtomicPtr, AtomicBool)` or `(*const T, bool)`.